opacus/optimizers/perlayeroptimizer.py

```diff
@@ -22,10 +21,16 @@ def __init__(
         assert len(max_grad_norms) == len(optimizer.params)
         self.max_grad_norms = max_grad_norms
         max_grad_norm = torch.norm(torch.Tensor(self.max_grad_norms), p=2)
-        super().__init__(optimizer, noise_multiplier=noise_multiplier, max_grad_norm=max_grad_norm, expected_batch_size=expected_batch_size, loss_reduction=loss_reduction)
+        super().__init__(
+            optimizer,
+            noise_multiplier=noise_multiplier,
+            max_grad_norm=max_grad_norm,
+            expected_batch_size=expected_batch_size,
+            loss_reduction=loss_reduction,
+        )
 
     def attach(self, optimizer):
-       self.optimizer = optimizer
+        self.optimizer = optimizer
```

Comment:
Not an issue, but do I understand correctly that if we have `optimizer = DPPerLayerOptimizer()` then the original optimizer is `optimizer.optimizer.optimizer`? 😆

Reply:
I wanted to ask though, is it a good pattern to inherit a class and store the parent's instance in the child's fields? Probably it is. Although I have a feeling that it may be possible to simplify by share fields with the parent and use super() calls.

Reply:
Way ahead of you: https://github.com/pytorch/opacus/blob/beeac9b50d59ea6304b7fb5be62d3d08f6e5ad99/opacus/privacy_engine.py#L232

I mean, yeah, we can do better at naming here.

But on a serious note, I'd love to have this as a proper inheritance, but I couldn't find a way.
Original optimizer can be of any `Optimizer` subclass, which makes it hard to handle this through inheritance